core/src/main/java/com/linecorp/armeria/common/metric/EventLoopMetrics.java

```diff
@@ -0,0 +1,134 @@
+import io.netty.util.concurrent.EventExecutor;
+
+/**
+ * A {@link MeterBinder} to observe Netty {@link EventLoopGroup}s. The following stats are currently
+ * exported per registered {@link MeterIdPrefix}.
+ *
+ * <ul>
+ *   <li>"event.loop.num.workers" (gauge) - the total number of Netty's event loops</li>
+ *   <li>"event.loop.pending.tasks" (gauge)
+ *     - the total number of IO tasks waiting to be run on event loops</li>
+ * </ul>
+ **/
+final class EventLoopMetrics implements MeterBinder {
+
+    private static final long UPDATE_INTERVAL_NANOS = TimeUnit.SECONDS.toNanos(3);
+
+    private final MeterIdPrefix idPrefix;
+    private final EventLoopGroup eventLoopGroup;
+
+    /**
+     * Creates an instance of {@link EventLoopMetrics}.
+     */
+    EventLoopMetrics(String name, EventLoopGroup eventLoopGroup) {
+        idPrefix = new MeterIdPrefix("armeria.netty").append(requireNonNull(name, "name"));
+        this.eventLoopGroup = requireNonNull(eventLoopGroup, "eventLoopGroup");
+    }
+
+    @Override
+    public void bindTo(MeterRegistry registry) {
+        final Self metrics = MicrometerUtil.register(registry, idPrefix, Self.class, Self::new);
+        metrics.add(eventLoopGroup);
+    }
+
+    /**
+     * An actual implementation of {@link EventLoopMetrics}.
+     */
+    static final class Self {
+        private final Set<EventLoopGroup> registry = ConcurrentHashMap.newKeySet(2);
+        private final Ticker ticker;
+
+        private volatile double lastPendingTasks;
+        private volatile double lastNumWorkers;
+        private volatile long lastStatsUpdateTime;
+
+        Self(MeterRegistry parent, MeterIdPrefix idPrefix) {
+            this(parent, idPrefix, Ticker.systemTicker());
+        }
+
+        Self(MeterRegistry parent, MeterIdPrefix idPrefix, Ticker ticker) {
+            this.ticker = requireNonNull(ticker, "ticker");
+
+            final String numWorkers = idPrefix.name("event.loop.workers");
+            parent.gauge(numWorkers, idPrefix.tags(), this, Self::numWorkers);
+
+            final String pendingTasks = idPrefix.name("event.loop.pending.tasks");
+            parent.gauge(pendingTasks, idPrefix.tags(), this, Self::pendingTasks);
+        }
+
+        // This runs every 3 seconds at most.
+        private void recomputeStatsIfNeeded() {
+            final long currentTimeNanos = ticker.read();
+            if (currentTimeNanos - lastStatsUpdateTime < UPDATE_INTERVAL_NANOS) {
+                return;
+            }
+
+            int numWorkers = 0;
+            int pendingTasks = 0;
+            for (EventLoopGroup group : registry) {
+
+                // Purge event loop groups that were shutdown.
+                if (group.isShutdown()) {
+                    registry.remove(group);
+                    continue;
+                }
+
+                for (EventExecutor eventLoop : group) {
+                    if (eventLoop instanceof SingleThreadEventLoop) {
```

Comment:
Why does it matter?

Reply:
The reasoning being in case users would like to also collect metrics for EventExecutors which aren't bound to a channel. (e.g. `DefaultEventExecutor`).
I think this scenario can be more prevalent when https://github.com/line/armeria/issues/4099 is handled.

It also seems like the finagle implementation you mentioned uses `SingleThreadEventExecutor` as well.

Reply:
> It also seems like the finagle implementation you mentioned uses SingleThreadEventExecutor as well.

Yep, but mostly because I wrote the original version of it and nobody ever touched it years.

Reply:
I see. I'm not particularly strong on this issue and we can always change it later if needed. Sounds good to me 👍